src/cache/impl/LevelDbCache.ts

```diff
@@ -22,19 +22,32 @@ export class LevelDbCache<V = any> implements SortKeyCache<V> {
    * and there doesn't seem to be any public interface/abstract type for all Level implementations
    * (the AbstractLevel is not exported from the package...)
    */
-  private readonly db: MemoryLevel;
+  private _db: MemoryLevel;
 
-  constructor(cacheOptions: CacheOptions) {
-    if (cacheOptions.inMemory) {
-      this.db = new MemoryLevel({ valueEncoding: 'json' });
+  // Configuration kept to initialize the cache upon first access
+  private readonly cacheOptions: CacheOptions;
+
+  // Lazy initialization upon first access
+  public get db(): MemoryLevel {
+    if (!!this._db) {
```

Comment:
usually in such cases I prefer this flow:

```ts
if (!this._db) {
  // init shit
}

return this._db;
```

\- single return point from the function

Reply:
Changed, I was minimizing nested ifs